CMakeLists.txt

```diff
@@ -64,11 +68,25 @@ endif()
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
+if(MSVC AND IGNORE_PYTHON_DEBUG_LIBRARY)
+  # A typical build on Windows needs to have Release and Debug libraries but most
+  # people don't have the python debug library. Hence, we ignore it here and link
+  # against the Release python library even in debug mode.
+  set(PYTHON3_DEBUG_LIBRARY_NAME python${Python3_VERSION_MAJOR}${Python3_VERSION_MINOR}_d.lib)
```

Comment:
Well, the quesion may be that why do we need Python for Libtorchvision? It is better to introduce a new variable `BUILD_PYTHON` and stop using those tricks.

Reply:
We shouldn't need Python for torchvision and we would love to remove it.

It is currently required only on Windows builds as we need https://github.com/pytorch/vision/blob/ce342580f3ae3f937fa5389c48b82a827b4804df/torchvision/csrc/vision.cpp#L13-L20 , but I think this might be an artifact of us using setuptools to compile the C++ extensions.

@datumbox has tried removing this recently to see if it was still needed, but Windows builds stopped working.

Do you have any ideas on how we could remove this Python dependency on Windows once and for all?

Reply:
@fmassa We can pass in a variable `USE_PYTHON`. Apparently, this function is unneeded for non-python builds.

Reply:
@peterjc123 Just to make sure I understand what you suggest:

- Add a CMake option USE_PYTHON which defines a preprocessor macro USE_PYTHON or WITH_PYTHON that would not be defined for C++-only builds done via CMake.
- That preprocessor macro would be defined when building with setuptools

If this is what you suggest I can try to add it to this PR.

Reply:
@willyd Yeah, that's exactly what I mean.

Reply:
@willyd FYI I just pushed into your branch to resolve a conflict related to a recent refactoring we did to the C++ code-base. 

Let me know if you have any questions concerning implementing the `USE_PYTHON` flag discussed above as we are still interested in receiving a PR that implements it.

Reply:
@datumbox  Thanks for letting me know. Yes I still intend to give it a shot. I am very busy at work lately, but I should have time to put on this PR before the end of the year.

Reply:
I tested the integration of `WITH_PYTHON` variable to the CMake project and it works in C++. I still need to rebase on master to get the latest changes and test again. It adds quite a few #ifdef so it is probably best to add a new header to avoid those. I will update this PR when I have something that works with the latest master.

Reply:
@willyd that would be great, thanks. Please ping me once you complete the PR to make sure we see it. Since this is quite old, it's in risk of missing any updates.

Reply:
@willyd Hi there! Wondering if we can close this PR or if you are still interested in sending the patch. Let me know, thanks! :)

Reply:
@datumbox Unfortunately, I don't have time to work on the patch. Let's close the PR and if I still have problems when I get back to it, I will submit a new PR.